segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -339,8 +339,6 @@ private void doStop(Throwable cause) {
                     Throwable failureCause = getFailureCause(this.durableLog, this.writer, this.metadataCleaner);
                     if (failureCause == null) {
                         failureCause = cause;
-                    } else if (cause != null && failureCause != cause) {
```

Comment:
When I was debugging some of the tests this error path caused a ref cycle which would in a different part of the code caused a OutOfStackSpace error because it sent error message processing into an infinite loop. 
I am not aware of this having been encountered anywhere else.